adam-core/src/main/scala/org/bdgenomics/adam/converters/AlignmentRecordConverter.scala

```diff
@@ -42,13 +47,8 @@ class AlignmentRecordConverter extends Serializable {
     val readNameSuffix =
       if (maybeAddSuffix &&
         !AlignmentRecordConverter.readNameHasPairedSuffix(adamRecord) &&
-        adamRecord.getFirstOfPair != null) {
-        "/" + (
-          if (adamRecord.getFirstOfPair)
-            "1"
-          else
-            "2"
-        )
+        adamRecord.getReadPaired) {
+        "/%d".format(adamRecord.getReadNum + 1)
       } else {
         ""
       }
```

Comment:
The quality score code isn't changed by this PR, so I'd prefer to have modifications to resolve #818 come in as a separate change. Additionally, since this is a pretty big PR already, I thought it'd be a bit easier to follow if side issues were tracked in separate tickets.